discovery/gossiper.go

```diff
@@ -1231,13 +1233,46 @@ func (d *AuthenticatedGossiper) retransmitStaleChannels(now time.Time) error {
 		signedUpdates = append(signedUpdates, chanUpdate)
 	}
 
-	// If we don't have any channels to re-broadcast, then we'll exit
+	// We'll also check that our NodeAnnouncement is not too old.
+	currentNodeAnn, err := d.cfg.SelfNodeAnnouncement(false)
+	if err != nil {
+		return fmt.Errorf("unable to get current node announment: %v",
+			err)
+	}
+
+	timestamp := time.Unix(int64(currentNodeAnn.Timestamp), 0)
+	timeElapsed := now.Sub(timestamp)
+
+	// If it's been a full day since we've re-broadcasted the
+	// node announcement, refresh it and resend it.
+	// TODO(halseth): only send if we have public channels.
```

Comment:
should this be a part of this pr? I think we can use the `IsPublicNode` method used below?

Reply:
Figured it was even simpler to just add a bool: https://github.com/lightningnetwork/lnd/pull/2084/commits/512efc855aae68a0b28c5544e34e4848b2642cc5